src/main/webapp/app/iris/settings/iris-settings-update/iris-common-sub-settings-update/iris-common-sub-settings-update.component.html

```diff
@@ -43,6 +43,40 @@ <h4 class="form-label mt-3" jhiTranslate="artemisApp.iris.settings.subSettings.e
         </div>
     </div>
 }
+@if (subSettings?.type === CHAT) {
+    <div class="mt-3">
+        <span class="form-label fs-4" jhiTranslate="artemisApp.iris.settings.subSettings.proactivitySettings.title"></span>
+        <fa-icon [icon]="faQuestionCircle" [ngbTooltip]="'artemisApp.iris.settings.subSettings.proactivitySettings.tooltip' | artemisTranslate" />
+    </div>
+    <div class="form-group">
+        <div class="form-group">
+            @for (event of exerciseChatEvents; track event) {
+                <div class="form-check form-switch mt-2 flex items-center">
+                    <input
+                        class="form-check-input"
+                        type="checkbox"
+                        id="{{ event }}"
+                        [disabled]="!isAdmin && !parentSubSettings"
+                        [ngModel]="!subSettings?.disabledProactiveEvents?.includes(event)"
+                        (ngModelChange)="onEventToggleChange(event)"
+                    />
+                    <label class="form-check-label" for="{{ event }}" [jhiTranslate]="eventTranslationKeys[event]"></label>
+                    @if (
+                        this.parentSubSettings &&
+                        !subSettings?.disabledProactiveEvents?.includes(event) &&
+                        (this.parentSubSettings.disabledProactiveEvents?.includes(event) || !this.parentSubSettings.enabled)
+                    ) {
```

Comment:
would it be possible to move this check out of the html? Having several function calls in here isn't best practice, as they are often re-evaluated

Reply:
I have moved the logic into the `ngOnChanges` life-cycle hook.